src/atomate2/openmm/jobs/base.py

```diff
@@ -0,0 +1,528 @@
+        )
+        platform_properties = self._resolve_attr("platform_properties", prev_task)
+
+        return interchange.to_openmm_simulation(
+            integrator,
+            platform=platform,
+            platformProperties=platform_properties,
+        )
+
+    def _update_interchange(
+        self,
+        interchange: Interchange,
+        sim: Simulation,
+        prev_task: OpenMMTaskDocument | None = None,
+    ) -> None:
+        """Update the Interchange object with the current simulation state.
+
+        This method updates the positions, velocities, and box vectors of the
+        Interchange object based on the current state of the OpenMM simulation.
+
+        Parameters
+        ----------
+        interchange : Interchange
+            The Interchange object to update.
+        sim : Simulation
+            The OpenMM simulation object.
+        prev_task : Optional[OpenMMTaskDocument]
+            The previous task document.
+        """
+        state = sim.context.getState(
+            getPositions=True,
+            getVelocities=True,
+            enforcePeriodicBox=self._resolve_attr("wrap_traj", prev_task),
+        )
+        if isinstance(interchange, Interchange):
+            interchange.positions = state.getPositions(asNumpy=True)
+            interchange.velocities = state.getVelocities(asNumpy=True)
+            interchange.box = state.getPeriodicBoxVectors(asNumpy=True)
+        elif isinstance(interchange, OpenMMInterchange):
+            interchange.state = XmlSerializer.serialize(state)
+
+    def _create_task_doc(
+        self,
+        interchange: Interchange,
+        elapsed_time: float | None = None,
+        dir_name: Path | None = None,
+        prev_task: OpenMMTaskDocument | None = None,
+    ) -> OpenMMTaskDocument:
+        """Create a task document for the OpenMM job.
+
+        This method creates an OpenMMTaskDocument based on the current
+        maker attributes, previous task document, and simulation results.
+
+        Parameters
+        ----------
+        interchange : Interchange
+            The updated Interchange object.
+        elapsed_time : Optional[float]
+            The elapsed time of the simulation. Default is None.
+        dir_name : Optional[Path]
+            The directory where the output files are saved.
+            Default is None.
+        prev_task : Optional[OpenMMTaskDocument]
+            The previous task document. Default is None.
+
+        Returns
+        -------
+        OpenMMTaskDocument
+            The created task document.
+        """
+        maker_attrs = copy.deepcopy(dict(vars(self)))
+        job_name = maker_attrs.pop("name")
+        tags = maker_attrs.pop("tags")
+        state_file_name = self._resolve_attr("state_file_name", prev_task)
+        traj_file_name = self._resolve_attr("traj_file_name", prev_task)
+        traj_file_type = self._resolve_attr("traj_file_type", prev_task)
+        calc = Calculation(
```

Comment:
Ideally the sub task documents should be created in a function in `OpenMMTaskDocument`. E.g., `OpenMMTaskDocument.from_directories()`. Can you move this logic there? The role of the `_create_task_doc` would then be to just resolve the previous directory names etc.

Reply:
I considered doing this originally but I think it'd be a headache. We could only reconstitute the information purely from the directories if we already had the current `task_doc` which is what we need to write. I'm sure there is some hack around but ultimately I feel this is where the logic belongs.

OpenMM has no input sets so the input logic is never fully encoded as a string. To get the task doc from a directory, we'd almost need to already have the taskdoc in the directory.